### PR TITLE
fix(admin-ui): keep user on Installs & Removes after manual return

### PR DESCRIPTION
**Pull request by dirkwa**
## Motivation

Reported by a user: *"App Store: once you have installed some updates, you can't get back to the 'Installs & Removes' tab, it just shows the 'Installed' tab."*

## Root cause

The auto-jump effect that lands the user on Installed when the install queue drains had `view` in its dep array and re-fired on every render where the queue was idle. Because `appStore.installing` is never cleared server-side (it drives the "Pending restart" warning), once any install had completed during the server's lifetime the user was bounced back to Installed every time they clicked the Installs & Removes tab.

## Fix

Detect the busy→idle transition via a `useRef` so the auto-jump fires once per completion rather than continuously while the queue is idle. `view` is dropped from the dep array so clicking the tab button no longer re-triggers the effect. The "auto-jump on completion" UX is preserved; the user can freely navigate back to Installs & Removes afterward; a subsequent install re-arms the detector naturally.

## Tested

- 5 new Vitest cases in `Apps.test.tsx` covering auto-jump on completion, sticky manual return, re-arm after a second install, failure-suppresses-jump, and stale-installing-on-mount.
- Full admin-ui build green; 215/215 admin-ui tests pass; root tsc build green.

## Summary

This PR fixes a tab-trapping issue in the Admin UI app store where the UI auto-jumps users back to the "Installed" tab whenever installs complete, preventing them from remaining on "Installs & Removes".

## Problem

The navigation effect auto-switches based on the install queue state and includes `view` in its dependency array. Because the server preserves `appStore.installing` (used for "Pending restart"), the effect re-fires while idle and repeatedly moves users to "Installed" when they click back to "Installs & Removes". Additionally, the effect does not account for failed installs.

## Solution

The component tracks the busy→idle transition using a ref and updates `view` only when the install queue transitions from busy to idle. The effect omits `view` from its dependencies to avoid retriggering on tab clicks. The auto-jump is suppressed if any install in the batch failed. Subsequent installs re-arm the transition detector so the behavior resumes on new completions.

## Changes

- packages/server-admin-ui/src/views/appstore/Apps/Apps.tsx
  - Use `useRef` to track previous busy count and detect busy→idle transitions.
  - Replace the previous effect with a transition-aware effect that excludes `view` from deps and checks for failures before auto-jumping.

- packages/server-admin-ui/src/views/appstore/Apps/Apps.test.tsx
  - Add 5 Vitest + React Testing Library tests covering:
    - Auto-jump on completion
    - Sticky manual return after an auto-jump
    - Re-arming after a subsequent install
    - Suppression of auto-jump on failure
    - Handling of stale `installing` state on mount

## Tests

- Admin-ui tests: 215/215 pass.
- Root TypeScript build: green.

### CODE REVIEW

**Comment by dirkwa** [comment]
seems PR2708 introduced a error here, investigating

**Comment by dirkwa** [comment]
ready for human review

**Comment by tkurki** [comment]
Needs rebase?

File: packages/server-admin-ui/src/views/appstore/Apps/Apps.tsx [hunk 1]
```diff
@@ -270,24 +271,27 @@ const Apps: React.FC = () => {
...
-      !hasFailure &&
-      appStore.installing.length > 0
-    ) {
+    const prev = prevBusyCountRef.current
+    prevBusyCountRef.current = busyCount
+    if (prev > 0 && busyCount === 0 && !hasFailure) {
       setSelectedView('Installed')
     }
-  }, [view, busyCount, hasFailure, appStore.installing.length])
+  }, [busyCount, hasFailure])
```
[the remark this answers is not in the document]
Comment (dirkwa) [reply packages/server-admin-ui/src/views/appstore/Apps/Apps.tsx:294]: @coderabbitai The whole-array failure scan is intentional, not a permanent block.

`installFailed` is re-derived from `modulesInstalledSinceStartup[module].code !== 0` on every install completion, so auto-jump re-arms as soon as no `installFailed` entries remain (e.g. after a successful retry) — as your own analysis notes. The current behaviour matches the documented intent in the comment block above the effect: suppress the auto-jump so the red "Install failed" pill stays in view until the user navigates away themselves, and there's a test asserting exactly that batch-level behaviour.

Scoping the check to a `prevInstallingIdsRef` snapshot of the specific busy→idle transition would add state to chase an edge case that isn't a regression, against the repo's YAGNI/KISS guidance. Leaving as-is.

**Comment by dirkwa** [comment]
ready for human review
